app/services/builders/external_pull_request.rb

```diff
@@ -0,0 +1,18 @@
+module Builders
+  class ExternalPullRequest < BaseService
```

Comment:
i wasn't sure when deciding that, i tested the classes around this "execution" but then i got into some redundance when i was going to test this, would you test the creation or query for the model ?

Reply:
Yeah, I know that feeling about redundancy haha
I tend to prefer that redundancy anyway, though. Because this class should be a standalone one and not necessarily be coupled with the other one.
As for this particular case, you could test the external pull request is created if not present and just returned if present. The related class that uses this one, `Processors::External:: PullRequests`, performs other checks and processes different payloads, so even though they are similar, they are not exactly the same tests.

Reply:
my criteria is something like:
- each class should be thoroughly tested with unit tests. this mean that unit tests should test a lot of different cases, including border cases
- integration tests should test only the most important flows and might check a reduced set of things, their focus is testing that some action is accomplished correctly